### PR TITLE
Add Compatibility Version

### PR DESCRIPTION
**Pull request by hlineholm-flir**
This is a basic functionality of a framework and it is required for it to function in a dependency graph. If the compatibility doesn't exist or is set to 0 you'll get:

`
dyld: Library not loaded: @rpath/SnapKit.framework/SnapKit
  Referenced from: /private/var/containers/Bundle/Application/2EAB6DCA-EF3C-461B-97FD-1666A542B2E7/XXX.app/Frameworks/XXX.framework/XXX
  Reason: Incompatible library version: XXX requires version 1.0.0 or later, but SnapKit provides version 0.0.0
`

As you can see, SnapKit tells **dylid** that it is compatible with version 0.0.0 which is not true.

### CODE REVIEW

**Comment by robertjpayne** [comment]
@hlineholm @LinusU this needs to be rebased against develop not master otherwise it's hard to decipher the changes at the moment

**Comment by LinusU** [comment]
#559 has already been merged already and no longer needs to be in here :)

File: Source/ConstraintConstantTarget.swift [hunk 1]
```diff
@@ -91,53 +95,73 @@ extension ConstraintConstantTarget {
...
                     return -value.right
-                case .bottom, .bottomMargin:
+                case .bottom, .bottomMargin, .lastBaseline:
                     return -value.bottom
                 case .leading, .leadingMargin:
                     return (ConstraintConfig.interfaceLayoutDirection == .leftToRight) ? value.left : value.right
                 case .trailing, .trailingMargin:
                     return (ConstraintConfig.interfaceLayoutDirection == .leftToRight) ? -value.right : -value.left
+                case .centerX, .centerXWithinMargins:
+                    return (value.left - value.right) / 2
```
Comment (colinta) [inline Source/ConstraintConstantTarget.swift]: Seems like this should be `(value.left + value.right) / 2` – the average, not the difference...?

Comment (colinta) [reply Source/ConstraintConstantTarget.swift]: Ohh no I see, the bottom and right values are inverted sign, so this *is* doing addition.  Got it.

**Comment by robertjpayne** [comment]
Believe this is obsolete now

**Pull request closed without merging**